src/Maestro/SubscriptionActorService/LocalGit.cs

```diff
@@ -46,40 +45,53 @@ public async Task<string> GetPathToLocalGitAsync()
             if (!string.IsNullOrEmpty(_gitExecutable) && File.Exists(_gitExecutable))
             {
                 _logger.LogInformation($"Git executable found at {_gitExecutable}. Checking the installation.");
-                // We should also mke sure that the git executable that exists runs properly
+                // We should also make sure that the git executable that exists runs properly
                 try
                 {
                     LocalHelpers.CheckGitInstallation(_gitExecutable, _logger);
                     return _gitExecutable;
                 }
                 catch
-                { 
+                {
                     _logger.LogWarning($"Something went wrong with validating git executable at {_gitExecutable}. Downloading new version.");
                 }
             }
 
-            await _semaphoreSlim.WaitAsync();
+            // Since the collisions are already using paths from _tempFiles.GetFilePath (instance-specific) we'll try
+            // a deterministic modification of that for the name of the semaphore (using a raw path as the name throws)
+            string semaphoreName = _tempFiles.GetFilePath("git-download-semaphore").Replace('/', '-').Replace('\\', '-').Replace(':', '-');
+            Semaphore crossProcessSemaphore = new Semaphore(1, 1, semaphoreName);
+
             try
             {
-                // Determine whether another thread ended up getting the lock and downloaded git
-                // in the meantime.
-                if (string.IsNullOrEmpty(_gitExecutable) || !File.Exists(_gitExecutable))
+                crossProcessSemaphore.WaitOne(TimeSpan.FromMinutes(5));
```

Comment:
What happens if this 5 minute timeout happens?  It's going to enter the finally and release a thing we haven't acquired, causing problems.

In general, the "lock a thing" should be outside the try, since if you fail to acquire the lock, you shouldn't release it.

Reply:
The generall pattern is
```
var a = GetAResource();
try {
} finally {
a.Release():
}
```

I usually end up writing IDiposable wrappers for things, but we dont' have one for non-slim (fat?) Semaphore.

Reply:
Oops, this was relying on the Mutex() behavior that throws if it fails to acquire.